### ISSUE TITLE
Issue with function arguments

### ISSUE THREAD

**Issue by henry2004y** [JuliaInterop/MATLAB.jl#161]
I have encountered an issue using `boundary` function from Matlab.

```
using MATLAB
x = rand(30);
y = rand(30);
k = mxcall(:boundary, 2, x, y) # This works fine
k = mxcall(:boundary, 3, x, y, 1.0) # This would return error
```

The error messages are:
```
Error using boundary (line 54)
Too many output arguments.
 
Error using save
Variable 'jx_boundary_arg_out_1' not found.
 
ERROR: MATLAB.MEngineError("failed to get variable jx_boundary_arg_out_1 from MATLAB session")
Stacktrace:
 [1] get_mvariable(::MSession, ::Symbol) at /Users/hyzhou/.julia/packages/MATLAB/cVrxc/src/engine.jl:164
 [2] mxcall(::MSession, ::Symbol, ::Int64, ::Array{Float64,1}, ::Vararg{Any,N} where N) at /Users/hyzhou/.julia/packages/MATLAB/cVrxc/src/engine.jl:297
 [3] mxcall(::Symbol, ::Int64, ::Array{Float64,1}, ::Array{Float64,1}, ::Vararg{Any,N} where N) at /Users/hyzhou/.julia/packages/MATLAB/cVrxc/src/engine.jl:317
 [4] top-level scope at none:0
```

My feeling is that the `boundary` function accepts different sets of input arguments which confuses Julia. How can we solve this?

**Comment by tqml** [comment]
Hey @henry2004y 

The issue seems to be that `k = mxcall(:boundary, 3, x, y, 1.0)` is incorrect. The `boundary` function in MATLAB only has at most 2 outputs (not 3). 

```
k = mxcall(:boundary, 2, x, y, 1.0) # works fine
```

I'm closing this issue, since it is seems to be incorrect use of the `boundary` function. Let me know if thats not the issue.

**Comment by henry2004y** [comment]
Thanks for the reply! I think I was confused about the meaning of the second argument in `mxcall` back 5 years ago and thought it refers to the number of input arguments.

**Comment by tqml** [comment]
Confusion was definitely justified, this function (among others) could need some docstrings and maybe better error messages

**Comment by henry2004y** [comment]
> Unfortunately no, I didn't know this function until today. But to me, the julia community seems super active so I would be suprised if there is no julia package that has something equivalent.

I would also be surprised, but sometimes it is really hard to find the right one since most of these wonderful works are buried under some packages, which makes them hard to search.

**Comment by tqml** [comment]
Indeed, maybe that could be solved with a global Julia documentation database and some kind of similarity search / LLM lookup
